src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -209,4 +226,66 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
             this.chart.remove();
         }
     };
+
+    private handlePushpinPositionChanged = (e: mapboxgl.EventData): void => {
+        const { target, type } = e;
+        const { onZoomChanged, onCenterPositionChanged } = this.props;
+        const zoom = this.getZoom(target.getZoom());
+        const center = this.getCenterPosition(target.getCenter());
+        const tooltipText = this.getTooltipText();
+        const supportedProperties: IGeoChartProperties = {
+            properties: {
+                controls: {
+                    ...zoom,
+                    ...center,
+                    ...tooltipText,
+                },
+            },
+        };
+
+        switch (type) {
+            case "zoomend":
+                onZoomChanged(supportedProperties);
+            case "moveend":
+                onCenterPositionChanged(supportedProperties);
+        }
+    };
+
+    private getCenterPosition = (center: IMapboxLngLat): IGeoCenterConfig | {} => {
+        const lng = center.lng;
+        const lat = center.lat;
+        // center format in properties should be [lat, lng]
+        const centerProp = transformPositionGeoConfig([lng, lat]);
+
+        if (
+            parseInt(lng.toString(), 10) === DEFAULT_LONGITUDE &&
+            parseInt(lat.toString(), 10) === DEFAULT_LATITUDE
+        ) {
+            return {};
+        }
+
+        return {
+            center: centerProp,
+        };
+    };
+
+    private getZoom = (zoom: number): IGeoZoomConfig | {} => {
+        return zoom !== DEFAULT_ZOOM ? { zoom } : {};
+    };
+
+    private getTooltipText(): IGeoTooltipText | {} {
+        const {
+            config: {
+                mdObject: { properties },
+            },
+        } = this.props;
+        if (!properties) {
+            return {};
+        }
+        const {
+            controls: { tooltipText },
+        } = JSON.parse(properties);
```

Comment:
should do `try/catch` for `JSON.parse` :)
btw, could `mdObject` be `undefined`?

Reply:
updated, change logic, so It's not necessary